js_modules/dagit/packages/core/src/plugins/ipynb.tsx

```diff
@@ -16,42 +16,56 @@ export const SidebarComponent: React.FC<IPluginSidebarProps> = (props) => {
 
   const metadata = props.definition.metadata;
   const notebookPath = metadata.find((m) => m.key === 'notebook_path');
+  const notebookKind = metadata.find((m) => m.key === 'kind');
   const repoLocName = props.repoAddress?.location;
 
   if (!notebookPath) {
     return <span />;
   }
 
-  return (
-    <div>
-      <Button icon={<Icon name="content_copy" />} onClick={() => setOpen(true)}>
-        View Notebook
-      </Button>
-      <Dialog
-        icon="info"
-        onClose={() => setOpen(false)}
-        style={{width: '80vw', maxWidth: 900}}
-        title={notebookPath.value.split('/').pop()}
-        usePortal={true}
-        isOpen={open}
-      >
-        <DialogBody>
-          <iframe
-            title={notebookPath.value}
-            src={`${rootServerURI}/dagit/notebook?path=${encodeURIComponent(
-              notebookPath.value,
-            )}&repoLocName=${repoLocName}`}
-            sandbox=""
-            style={{border: 0, background: 'white'}}
-            seamless={true}
-            width="100%"
-            height={500}
-          />
-        </DialogBody>
-        <DialogFooter>
-          <Button onClick={() => setOpen(false)}>Close</Button>
-        </DialogFooter>
-      </Dialog>
-    </div>
-  );
+  if (notebookKind && notebookKind.value === 'noteable') {
+    return (
+      <div>
+        <Button
+          icon={<Icon name="content_copy" />}
+          onClick={() => window.open(notebookPath.value, '_blank')}
```

Comment:
I'm not familiar with the different types of notebook possibilities here. Is it always preferable to open Noteable notebooks in a new tab instead of the iframe dialog (as done below)? If were were able to determine whether the `value` is a URL, would it be better to use the iframe or a new tab?

Reply:
I think yes, it'd always be preferable to open a Noteable notebook (or other hosted notebook) in a new tab

Reply:
@yuhan do you have any thoughts on this? I don't see a case where we wouldn't want to open a noteable notebook in a new tab (vs rendering in dagit), but I may be missing something. And I think it makes sense that we would generalize the behavior so that if `notebookPath` is a URL then we open it in a new tab

Reply:
imo for integration, it'd be better to open in a new tab (it feels to me more like an impression/experience feeling that, for example, if anything goes wrong on the third party end, when it's in a separate tab, the user would know immediately where to seek for help, whereas if we embed it in an iframe the user may not realize it was outside dagster). but for plain notebooks, we'd still want to show them in iframes.

> And I think it makes sense that we would generalize the behavior so that if notebookPath is a URL then we open it in a new tab
i like this idea.

Reply:
ok! @hellendag do you know of a way for us to check if a string is a URL?

Reply:
- What does `value` look like in the iframe case? Is it a path or something?
- Is the URL expected to be an external URL?

If you want to check whether the value is a valid full URL, you could do something like:

```
const value = notebookPath.value;
const url = React.useMemo(() => {
  let urlString = null;
  try {
    const url = new URL(value);
    return url.toString();
  } catch (e) {
    // Not a full valid URL
    return null;
  }
}, [value]);

if (url) {
  return <ExternalAnchorButton href={url}>...</ExternalAnchorButton>;
}

// Otherwise, it's presumably a path to open in an iframe Dialog?
return <...>...</...>;
```